### PR TITLE
fix(noclasspath): NPE throws with annotations on primitive type.

### LINKED ISSUE #529: NPE in `JDTTreeBuilder.isContainsInTypeAnnotation`
I found another `NullPointerException` to be thrown by JDTTreeBuilder in noclassepath-mode:

```
Exception in thread "main" java.lang.NullPointerException
	at spoon.support.compiler.jdt.JDTTreeBuilder.isContainsInTypeAnnotation(JDTTreeBuilder.java:2733)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:2706)
	at org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.traverse(MethodDeclaration.java:327)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.traverse(TypeDeclaration.java:1415)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:2848)
	at org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression.traverse(QualifiedAllocationExpression.java:604)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:2122)
	at org.eclipse.jdt.internal.compiler.ast.Assignment.traverse(Assignment.java:262)
	at org.eclipse.jdt.internal.compiler.ast.Block.traverse(Block.java:139)
	at org.eclipse.jdt.internal.compiler.ast.SynchronizedStatement.traverse(SynchronizedStatement.java:212)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:2726)
	at org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.traverse(MethodDeclaration.java:327)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:3459)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.traverse(TypeDeclaration.java:1319)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.traverse(CompilationUnitDeclaration.java:748)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.traverse(CompilationUnitDeclaration.java:709)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.buildSources(JDTBasedSpoonCompiler.java:228)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:481)
```

This code triggers the exception (I don't know which part exactly):

```Java
import example.B;

class A extends B {

    private Runnable runnable;

    public void foo() {
        synchronized (lock) {
           runnable = new Runnable() {

                @Override
                public void run() {

                }
            };
        }
    }
}
```

I tested it with Spoon 5.0.1 and 5.0.2.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #529

### CODE REVIEW

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

File: src/main/java/spoon/reflect/declaration/CtType.java [hunk 1]
```diff
@@ -32,6 +32,10 @@
 	 * The string separator in a Java innertype qualified name.
 	 */
 	String INNERTTYPE_SEPARATOR = "$";
+	/**
+	 * Used in no classpath when we don't have any information to build the name of the type.
+	 */
+	String NAME_UNKNOWN = "<unknown>";
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtType.java]: put in a pubic so that client code can test against it

Comment (GerardPaligot) [inline src/main/java/spoon/reflect/declaration/CtType.java]: This is declared in an interface. It is public.

Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtType.java]: OK I'll merge when Travis is OK.
Travis is slow today.

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**